### ISSUE TITLE
Support web platform

### ISSUE THREAD

**Issue by dwieeb** [capacitor-community/fcm#46]
I noticed this plugin does not support web/PWA. FCM users can just use the Firebase client for JS, but for a better cross-platform experience, we can wrap the client in the existing API of this plugin.

@CFT-Chris Maybe you could chime in? I saw [your plugin](https://github.com/CFT-Chris/capacitor-pwa-firebase-msg) seems to provide the necessary functionality. Perhaps they could be merged?

**Comment by CFT-Chris** [comment]
Yes, I will gladly merge my [plugin](https://github.com/CFT-Chris/capacitor-pwa-firebase-msg) with this project.  I can work on that when there is a settled method of aggregating service worker code for Capacitor plugins, and when we can put the capacitor.plugin.json and service worker files separated from the www staging directory and into its own platform/web directory.

**Comment by dwieeb** [comment]
@CFT-Chris Want to chat on the [Capacitor Slack](https://getcapacitor.herokuapp.com/)?

**Comment by CFT-Chris** [comment]
@dwieeb Sure.  I joined the Slack, but couldn't find you in the search.  You can find me "Chris L" with the same blue Gravatar.

**Comment by Ka-Repo** [comment]
@CFT-Chris @imhoffd any update?

**Comment by akshitkrnagpal** [comment]
Are there any updates on this?

**Comment by CFT-Chris** [comment]
I won't have a chance to look into this for a while, so anyone else can feel free to start this work.

Porting my work from https://github.com/CFT-Chris/capacitor-pwa-firebase-msg is a possible starting point, though I haven't looked into v3 Capacitor and am not sure if they merge together very well.
